add bushwacka and darwin's revert

haaksirikko opened the pull request:
Bushwacka
  - Crits whenever it would normally mini-crit
  - 20% fire damage vulnerability on wearer

Darwin's Danger Shield
  - +25 max health on wearer
  - +15% bullet damage resistance on wearer
  - 20% explosive damage vulnerability on wearer

haaksirikko committed:
add bushwacka and darwin's revert

Bushwacka
  - Crits whenever it would normally mini-crit
  - 20% fire damage vulnerability on wearer

Darwin's Danger Shield
  - +25 max health on wearer

haaksirikko committed:
fix incorrect attrib value

antielectronbomb commented:
Works on Windows 10, for both the Bushwacka and the Darwin's Danger Shield without major issues.

If you want my opinion on this, both the release versions of Bushwacka and DDS needs the item set bonus (headshot immunity), especially for the Darwin's Danger Shield because it's just a flat HP buff. The old set bonus gave the Sniper a different playstyle (passive hardscoping playstyle that requires less skill instead of active quickscoping playstyle that requires more skill) in my opinion. Anyways this is for another topic better discussed in the Steam Group forums.

Although there is a really minor and esoteric issue that can be safely ignored where you get overhealed by a Medic while wearing DDS, then switch to another secondary while being healed, then your HP slowly drops to 187 HP instead of 185 HP due to rounding up shenanigans. This issue can be safely ignored.

rsedxcftvgyhbujnkiqwe commented:
An idea I'd be curious to float is to use the 2013 version of the danger shield which would make it a more "interesting" weapon to use than just a boring +25 hp

When item sets got removed in July of 2013, the danger shield received the following:
-Added Bullet damage taken reduced by 15%
-Added Explosive damage taken increased by 20%
This makes you able to survive sniper quickscopes (fully uncharged ones), but much more vulnerable to being bombed by soldiers and demos. Not being "immune" to headshots might make this more balanced as a weapon. Can discuss it more in the revert thread

Regardless I'll consider merging this soon and we can run it as a week long test to see how people like it.

antielectronbomb commented on scripting/reverts.sp:
```diff
@@ -1715,6 +1731,19 @@ public Action TF2Items_OnGiveNamedItem(int client, char[] class, int index, Hand
+	) {
+		item1 = TF2Items_CreateItem(0);
+		TF2Items_SetFlags(item1, (OVERRIDE_ATTRIBUTES|PRESERVE_ATTRIBUTES));
+		TF2Items_SetNumAttributes(item1, 3);
```
Since random wants to try the pre-Inferno Danger Shield, add the following:

```
		TF2Items_SetNumAttributes(item1, 5);
		TF2Items_SetAttribute(item1, 0, 60, 1.0); // +0% fire damage resistance on wearer
		TF2Items_SetAttribute(item1, 1, 527, 0.0); // remove afterburn immunity
		TF2Items_SetAttribute(item1, 2, 26, 25.0); // +25 max health on wearer
		TF2Items_SetAttribute(item1, 3, 66, 0.85); // +15% bullet damage resistance on wearer
		TF2Items_SetAttribute(item1, 4, 65, 1.20); // 20% explosive damage vulnerability on wearer
```

I've already tested these new attributes, works with bots.

antielectronbomb commented on scripting/reverts.sp:
```diff
@@ -300,13 +300,15 @@ public void OnPluginStart() {
 	ItemDefine("Cozy Camper","cozycamper","Reverted to pre-matchmaking, flinch resist at any charge level", CLASSFLAG_SNIPER);
 #endif
 	ItemDefine("Crit-a-Cola", "critcola", "Reverted to pre-matchmaking, +25% movespeed, +10% damage taken, no mark-for-death on attack", CLASSFLAG_SCOUT);
+	ItemDefine("Darwin's Danger Shield", "darwin", "Reverted to pre-2013, +25 max health, no afterburn immunity, no damage modifiers", CLASSFLAG_SNIPER);
```
Change to pre-Inferno stats
`ItemDefine("Darwin's Danger Shield", "darwin", "Reverted to pre-inferno, +25 max HP, no afterburn immunity, 15% bullet resist, 20% blast vuln", CLASSFLAG_SNIPER);`

antielectronbomb reviewed (commented):
Pre-Inferno Darwin's Danger Shield Damage Testing

```
+25 max HP works
Afterburn and fire resist tested

Old stats:
	+25 Max HP
	+15% bullet damage resistance on wearer
	20% explosive damage vulnerability on wearer
	No Afterburn immunity
	No +50% fire resist

BULLET DAMAGE TESTING:
	No Shield:
		Base: 50
		Minicrit: 65
		Crit: 150

	With Shield:
		Base: 43
		Minicrit: 60
		Crit: 143

	Bullet Resistance Percentages:
		Base: 15%
		Minicrit: 7.69%
		Crit: 4.67%

EXPLOSIVE DAMAGE TESTING:
	No Shield:
		Base: 100
		Minicrit: 135
		Crit: 300

	With Shield:
		Base: 120
		Minicrit: 155
		Crit: 320

	Explosive Vulnerability Percentages:
		Base: 20%
		Minicrit: 14.8%
		Crit: 6.67%
```
		
Conclusion:
-Resistance and damage modifiers are smaller than that for base damage
-Works on Windows, tested with bots (I had to use addcond 74 to give me 10x HP to test explosive damage)

haaksirikko committed:
pre-inferno danger shield

Darwin's Danger Shield
  - +25 max health on wearer
  - +15% bullet damage resistance on wearer
  - 20% explosive damage vulnerability on wearer
